### PR TITLE
Grf 705 : Delete template + attributes & link to category tree

### PR DESCRIPTION
**Pull request by jeremyBeaucousinAkeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/Sql/GetEnrichedValuesByTemplateUuidSqlIntegration.php [hunk 1]
```diff
@@ -63,4 +65,33 @@ public function testItRetrievesRelatedCategoriesByTemplateUuid(): void{
...
+        Assert::assertArrayHasKey('socks', $fetchedCategories[0]);
+        Assert::assertArrayHasKey('winter_socks', $fetchedCategories[0]);
+        Assert::assertArrayHasKey('summer_socks', $fetchedCategories[0]);
+        Assert::assertArrayHasKey('japanese_summer_socks', $fetchedCategories[0]);
+        Assert::assertArrayNotHasKey('winter_socks_null', $fetchedCategories[0]);
+        Assert::assertNotEmpty($fetchedCategories[0]['socks']->getValues());
+        Assert::assertNotEmpty($fetchedCategories[0]['winter_socks']->getValues());
+        Assert::assertNotEmpty($fetchedCategories[0]['summer_socks']->getValues());
+        Assert::assertNotEmpty($fetchedCategories[0]['japanese_summer_socks']->getValues());
+        Assert::assertCount(4, $fetchedCategories[0]);
```
Comment (amigaire) [inline src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/Sql/GetEnrichedValuesByTemplateUuidSqlIntegration.php:95]:
```suggestion
	Assert::assertCount(4, $sockCategoryEnrichedValues);

	$sockCategoryEnrichedValues = $fetchedCategories[0];

	Assert::assertArrayHasKey('socks', $sockCategoryEnrichedValues);
	Assert::assertArrayHasKey('winter_socks', $sockCategoryEnrichedValues);
	Assert::assertArrayHasKey('summer_socks', $sockCategoryEnrichedValues);
	Assert::assertArrayHasKey('japanese_summer_socks', $sockCategoryEnrichedValues);
	Assert::assertArrayNotHasKey('winter_socks_null', $sockCategoryEnrichedValues);
	Assert::assertNotEmpty($sockCategoryEnrichedValues['socks']->getValues());
	Assert::assertNotEmpty($sockCategoryEnrichedValues['winter_socks']->getValues());
	Assert::assertNotEmpty($sockCategoryEnrichedValues['summer_socks']->getValues());
	Assert::assertNotEmpty($sockCategoryEnrichedValues['japanese_summer_socks']->getValues());
```

File: src/Akeneo/Category/back/Application/Query/DeleteCategoryTreeTemplate.php [hunk 2]
```diff
@@ -12,5 +13,7 @@
  */
 interface DeleteCategoryTreeTemplate
 {
-    public function __invoke(int $categoryTreeId, TemplateUuid $templateUuid): void;
+    public function byCategoryIdAndTemplateUuid(CategoryId $categoryTreeId, TemplateUuid $templateUuid): void;
+
+    public function byTemplateUuid(TemplateUuid $templateUuid): void;
```
Comment (pchasle) [inline src/Akeneo/Category/back/Application/Query/DeleteCategoryTreeTemplate.php:18]: Can you create a separate query class for this?

Also, the new interface (`DeleteCategoryTemplate`?) can be in the domain right away, under `Domain/Query/`

Comment (pchasle) [reply src/Akeneo/Category/back/Application/Query/DeleteCategoryTreeTemplate.php:18]: To complete a little bit, I'm in favor of having one query per class only.

**Review by amigaire — APPROVED** [review]

File: src/Akeneo/Category/back/tests/Specification/Infrastructure/EventSubscriber/RemoveCategoryTreeTemplateSubscriberSpec.php [hunk 3]
```diff
@@ -24,10 +24,10 @@
 class RemoveCategoryTreeTemplateSubscriberSpec extends ObjectBehavior
 {
     function let(
-        GetCategoryInterface $getCategory,
-        GetCategoryTemplateByCategoryTree $getCategoryTemplateByCategoryTree,
-        DeleteCategoryTreeTemplate $deleteCategoryTreeTemplate,
-        FeatureFlag $enrichedCategoryFeature
+        GetCategoryInterface                                  $getCategory,
```
Comment (pchasle) [inline src/Akeneo/Category/back/tests/Specification/Infrastructure/EventSubscriber/RemoveCategoryTreeTemplateSubscriberSpec.php:27]: Indentation to fix?

**Pull request merged**